src/collectors/stdio/mod.rs

```diff
@@ -2,29 +2,52 @@ use std::io;
 use std::io::Write;
 use std::error::Error;
 use events;
+use formatters;
 
 pub struct StdioCollector {
-    _use_stderr: bool
+    use_stderr: bool,
+    formatter: Box<formatters::TextFormatter + Sync>
```

Comment:
What do you think the difference would be if `formatter` was a generic argument instead of a boxed value? Would the `StdioCollector<F>` because tedious to use in a pipeline?